source/mbed_trace.c

```diff
@@ -330,10 +378,11 @@ void mbed_vtracef(uint8_t dlevel, const char *grp, const char *fmt, va_list ap)
         mbed_trace_reset_tmp();
         goto end;
     }
-    if ((m_trace.trace_config & TRACE_MASK_LEVEL) &  dlevel) {
-        bool color = (m_trace.trace_config & TRACE_MODE_COLOR) != 0;
-        bool plain = (m_trace.trace_config & TRACE_MODE_PLAIN) != 0;
-        bool cr    = (m_trace.trace_config & TRACE_CARRIAGE_RETURN) != 0;
+    uint8_t config = mbed_trace_get_module_config(grp);
```

Comment:
In mbed-trace nomenclature, that should probably be "group config" instead of "module" config.